tests/unittests/audio/test_dnsmos.py

```diff
@@ -0,0 +1,257 @@
+            predicted_p808_mos.append(p808_mos)
+
+        return {
+            "len_in_sec": actual_audio_len / fs,
+            "sr": fs,
+            "num_hops": num_hops,
+            "OVRL_raw": np.mean(predicted_mos_ovr_seg_raw),
+            "SIG_raw": np.mean(predicted_mos_sig_seg_raw),
+            "BAK_raw": np.mean(predicted_mos_bak_seg_raw),
+            "OVRL": np.mean(predicted_mos_ovr_seg),
+            "SIG": np.mean(predicted_mos_sig_seg),
+            "BAK": np.mean(predicted_mos_bak_seg),
+            "P808_MOS": np.mean(predicted_p808_mos),
+        }
+
+
+def _reference_metric_batch(
+    preds: Tensor,
+    target: Tensor,
+    fs: int,
+    personalized: bool,
+    device: Optional[str] = None,
+    reduce_mean: bool = False,
+    **kwargs: Dict[str, Any],
+):
+    # shape: preds [BATCH_SIZE, Time]
+    # download onnx first
+    _load_session(f"{DNSMOS_DIR}/{'p' if personalized else ''}DNSMOS/sig_bak_ovr.onnx", torch.device("cpu"))
+    _load_session(f"{DNSMOS_DIR}/DNSMOS/model_v8.onnx", torch.device("cpu"))
+    # construct ComputeScore
+    cs = _ComputeScore(
+        f"{DNSMOS_DIR}/{'p' if personalized else ''}DNSMOS/sig_bak_ovr.onnx",
+        f"{DNSMOS_DIR}/DNSMOS/model_v8.onnx",
+    )
+
+    shape = preds.shape
+    preds = preds.reshape(1, -1) if len(shape) == 1 else preds.reshape(-1, shape[-1])
+    n_batch, time = preds.shape
+
+    preds = preds.detach().cpu().numpy()
+    score = []
+    for b in range(preds.shape[0]):
+        val = cs.__call__(preds[b, ...], fs, personalized)
+        score.append([val["P808_MOS"], val["SIG"], val["BAK"], val["OVRL"]])
+    score = torch.tensor(score)
+    if reduce_mean:
+        # shape: preds [BATCH_SIZE, 1, Time] , target [BATCH_SIZE, 1, Time]
+        # or shape: preds [NUM_BATCHES*BATCH_SIZE, 1, Time] , target [NUM_BATCHES*BATCH_SIZE, 1, Time]
+        return score.mean(dim=0)
+    return score.reshape(*shape[:-1], 4).reshape(shape[:-1] + (4,)).numpy()
+
+
+def _dnsmos_cheat(preds, target, **kwargs: Dict[str, Any]):
+    # cheat the MetricTester as the deep_noise_suppression_mean_opinion_score doesn't need target
+    return deep_noise_suppression_mean_opinion_score(preds, **kwargs)
+
+
+class _DNSMOSCheat(DeepNoiseSuppressionMeanOpinionScore):
+    # cheat the MetricTester as DeepNoiseSuppressionMeanOpinionScore doesn't need target
+    def update(self, preds: Tensor, target: Tensor) -> None:
+        super().update(preds=preds)
+
+
+seed_all(42)
+
+preds = torch.rand(2, 2, 8000)
+
+
+@pytest.mark.parametrize(
+    "preds, fs, personalized",
+    [
+        (preds, 8000, False),
+        (preds, 8000, True),
+        (preds, 16000, False),
+        (preds, 16000, True),
+    ],
+)
+class TestDNSMOS(MetricTester):
+    """Test class for `DeepNoiseSuppressionMeanOpinionScore` metric."""
+
+    atol = 5e-3
+
+    @pytest.mark.parametrize("ddp", [pytest.param(True, marks=pytest.mark.DDP), False])
+    def test_dnsmos(self, preds, fs, personalized, ddp, device=None):
```

Comment:
```suggestion
    def test_dnsmos(self, preds: Tensor, fs: int, personalized: bool, ddp: bool, device=None):
```

Same thing for methods below